rcl/src/rcl/log_level.c

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rcl/log_level.h"
+
+#include "rcl/error_handling.h"
+#include "rcutils/allocator.h"
+#include "rcutils/strdup.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#define MAX_NUM_LOGGER_SETTING_ENTRIES 256U
```

Comment:
We've been (slowly) trying to remove these arbitrary limits from the code.  Can you make it so that this isn't an arbitrary limit, but is just based on how many the user passed in?